src/raw_iter.rs

```diff
@@ -69,17 +69,372 @@ enum DynamicRawColors<'a> {
+        }
+    }
+
+    /// Bump the cursor to the next position in the bitmap.
+    ///
+    /// Note that RLE bitmaps are upside down.
+    fn move_position(&mut self) -> Point {
+        let old_position = self.next_pixel;
+        self.next_pixel.x += 1;
+        if self.next_pixel.x == self.width as i32 {
+            self.next_pixel.x = 0;
+            self.next_pixel.y = self.next_pixel.y.saturating_sub(1);
+        }
+        old_position
+    }
+}
+
+impl<'a> Iterator for Rle4Pixels<'a> {
+    type Item = (Point, u32);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            match self.rle_state {
+                RleState::EndOfBitmap => {
+                    return None;
+                }
+                RleState::Absolute {
+                    remaining,
+                    is_odd,
+                    has_padding,
+                } => {
+                    // Here `remaining` is a count of nibbles, not bytes
+
+                    // [00, 04, 45, 67] => 4 5 6 7
+                    //                     ^ ^ ^ ^
+                    //                     | | | +-- remaining is 0, is_odd is 0, we want [1]right
+                    //                     | | +-- remaining is 1, is_odd is 0, we want [1]left
+                    //                     | +-- remaining is 2, is_odd is 0, we want [0]right
+                    //                     +-- remaining is 3, is_odd is 0, we want [0]left
+                    // [00, 03, 45, 60] => 4 5 6
+                    //                     ^ ^ ^
+                    //                     | | +-- remaining is 0, is_odd is 1, we want [1]left
+                    //                     | +-- remaining is 1, is_odd is 1, we want [0]right
+                    //                     +-- remaining is 2, is_odd is 1, we want [0]left
+
+                    let remaining_is_odd = (remaining % 2) != 0;
+                    let want_left = remaining_is_odd != is_odd;
+
+                    if remaining == 0 {
+                        self.rle_state = RleState::Starting;
+                    } else {
+                        self.rle_state = RleState::Absolute {
+                            remaining: remaining.saturating_sub(1),
+                            is_odd,
+                            has_padding,
+                        };
+                    }
+
+                    let value = self.data.first()?;
+                    let nibble_value = if want_left { value >> 4 } else { value & 0x0F };
+                    if !want_left || remaining == 0 {
+                        self.data = self.data.get(1..)?;
+                    }
+                    if remaining == 0 && has_padding {
+                        // remove the padding byte too
+                        self.data = self.data.get(1..)?;
+                    }
+                    let this_pixel = self.move_position();
+                    return Some((this_pixel, nibble_value as u32));
```

Comment:
I prefer to use `u32::from`, because it makes clear that this is a save operation.
```suggestion
                    return Some((this_pixel, u32::from(nibble_value)));
```